xen: cleanup, code quality improvements

SigmaSquadron opened the pull request:
## Description of changes

Minor changes and a cherry-pick of #341429 in preparation for a larger PR that begins building the MiniOS stubdomains.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested, as applicable:
  I've been dogfooding this branch for a while now.
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 342915"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
  - Not sure. Let me know if the commits should be squashed.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SigmaSquadron committed:
xen: allow importing the entire meta section from versionDefinition

Also makes it slightly more readable by adding line breaks between
attributes and reordering the knownVulnerabilities attribute to be
closer to the maintainers attribute, as they have similar checks.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
xen: simplify inherit calls

Instead of using multiple inherit calls, just use a single one.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
xen: move from xen.org to xenproject.org

xenproject.org is the Xen Project's official domain. xen.org is
deprecated and we should avoid referring to Xen as just "xen" whenever
possible, due to a messy legal history with Citrix.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
xen: remove XSA_458 patch

None of our Xen branches need it anymore.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
xen: minor updates to the update script

- Drop `-ie` flags for grep;
- Add more `doing things...done` echo stanzas so it's clearer when a
part of the script has concluded;
- Add a note explaining why QEMU and OVMF take so long to fetch;
- Add a TODO for future work at the top of the file.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

emilazy commented on pkgs/applications/virtualization/xen/update.sh:
```diff
@@ -120,13 +127,13 @@ for version in "${supportedVersions[@]}"; do
 
     # Prepare patches that are called in ./patches.nix.
-    defaultPatchListInit=("QUBES_REPRODUCIBLE_BUILDS" "XSA_458" "XSA_460" "XSA_461" )
+    defaultPatchListInit=("QUBES_REPRODUCIBLE_BUILDS" "XSA_460" "XSA_461" )
```
What’s the deal with `XSA_458`?

SigmaSquadron replied:
Already included in 4.17.5, 4.18.3 and 4.19.0, so it's unused.

emilazy reviewed (commented):
Generally LGTM.

emilazy commented on pkgs/applications/virtualization/xen/generic/0000-xen-ipxe-src-generic.patch:
```diff
@@ -1,6 +1,6 @@
 Hack to make etherboot use pre-fetched iPXE.
 
-diff --git a/tools/firmware/etherboot/Makefile b/tools/firmware/etherboot/Makefile
+diff a/tools/firmware/etherboot/Makefile b/tools/firmware/etherboot/Makefile
```
What happened here? Looks like a Git format diff.

SigmaSquadron replied:
Just cosmetics. The flag doesn't really matter one way or the other, so I thought I'd just keep the path. Another patch by Lach had other flags that I also removed for consistency, so I standardised this one as well.

SigmaSquadron commented:
OfBorg is failing :(

SigmaSquadron commented:
Big surprise. I was stupid and forgot how newlines work.

SigmaSquadron committed:
xen: don't patch QEMU and OVMF if they aren't being built

This resolves the spurious 'directory not found' errors in the xen-slim
builds.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron commented:
Part II: #343111
